pkg/exec/builder/execute_test.go

```diff
@@ -156,3 +157,24 @@ func TestExecuteStep_SpecifiesCustomWorkingDirectory(t *testing.T) {
 	assert.Equal(t, fmt.Sprintln(wd), c.GetOutput())
 	require.NoError(t, err, "Execute Step failed")
 }
+
+func (s TestOrderedStep) GetEnvironmentVars() map[string]string {
+	return s.EnvironmentVars
+}
+
+func TestExecuteStep_WithEnvironmentVars(t *testing.T) {
+	c := context.NewTestContext(t)
+
+	step := TestOrderedStep{
+		TestStep: TestStep{
+			Command:         "env",
+			EnvironmentVars: map[string]string{"SOME_VAR_123": "foo"},
+		},
+	}
+
+	c.Setenv(test.ExpectedCommandEnv, "env")
+
+	_, err := ExecuteStep(c.Context, step)
+	require.NoError(t, err, "Execute Step failed")
+	assert.Contains(t, fmt.Sprintln("SOME_VAR_123=foo"), c.GetOutput())
```

Comment:
This assert isn't checking what you think here. It's saying assert "SOME_VAR_123=foo" contains "", which will always pass. After changing this line to the following:

```suggestion
	assert.Contains(t, c.GetOutput(), fmt.Sprintln("SOME_VAR_123=foo"))
```

the test fails because context.NewCommand is using a mock for unit tests and isn't actually calling env.

Here's how to get your env command to run:

```
// Skip on windows because env is mac/linux only
if runtime.GOOS == "windows" {
  t.Skip()
}

c := context.NewTestContext(t)
c.UseFilesystem() // Don't use a mocked file system or NewCommand function, actually run commands against the host
```